container_test.go

```diff
@@ -489,6 +490,53 @@ func TestShouldStartContainersInParallel(t *testing.T) {
 	}
 }
 
+func TestLabelBuilding(t *testing.T) {
+	labels := core.DefaultLabels("session-id")
+	assert.NotNil(t, labels)
+	assert.Len(t, labels, 4)
+	goodCustomLabels := map[string]string{
+		"org.custom.label1": "foo",
+		"org.custom.label2": "bar",
+	}
+	err := core.MergeCustomLabels(goodCustomLabels, labels)
+	require.NoError(t, err)
+	assert.NotNil(t, goodCustomLabels)
+	assert.Contains(t, goodCustomLabels, core.LabelLang)
+	assert.Len(t, goodCustomLabels, 6)
+
+	badCustomLabels := map[string]string{
+		core.LabelLang:      "python",
+		core.LabelSessionID: "invalid",
+	}
+	err = core.MergeCustomLabels(badCustomLabels, labels)
+	require.Error(t, err)
+}
+
+func TestLabelsForContainer(t *testing.T) {
+	ctx := context.Background()
+
+	customLabel := "org.company.environment"
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image:  "alpine:latest",
+			Labels: map[string]string{customLabel: "test"},
+		},
+	})
+	require.NoError(t, err)
+
```

Comment:
issue: `defer` the container terminate here, otherwise it won't get done on a test failure.
```suggestion
	require.NoError(t, err)

	t.Cleanup(func() {
		require.NoError(t, container.Terminate(context.Background()))
	})
```